src/base/utils/string.h

```diff
@@ -63,10 +65,39 @@ namespace Utils::String
 
     QStringList splitCommand(const QString &command);
 
-    QString join(const QList<QStringView> &strings, QStringView separator);
-
     QString fromDouble(double n, int precision);
 
+    template <typename Container>
+    QString joinIntoString(const Container &container, const QString &separator)
+        requires ExplicitlyConvertibleTo<typename Container::value_type, QString>
+    {
+        auto iter = container.cbegin();
+        const auto end = container.cend();
+        if (iter == end)
+            return {};
+
+        // Some `join` implementation will iterate over the elements to count the total size and other
```

Comment:
What kind of implementations are you talking about here? I don't understand the purpose of this comment.

Reply:
The comment isn't relevant now. Removed.